docs/excel/excel-add-ins-ranges-advanced.md

```diff
@@ -0,0 +1,166 @@
+
+Range’s `copyFrom` function replicates the copy-and-paste behavior of the Excel UI. The range object that `copyFrom` is called on is the destination.
+The source to be copied is passed as a range or a string address representing a range. 
+The following code sample copies the data from **A1:E1** into the range starting at **G1** (which ends up pasting into **G1:K1**).
+
+```js
+Excel.run(function (context) {
+    var sheet = context.workbook.worksheets.getItem("Sample");
+    // copy a range starting at a single cell destination
+    sheet.getRange("G1").copyFrom("A1:E1");
+    return context.sync();
+}).catch(errorHandlerFunction);
+```
+
+`Range.copyFrom` has three optional parameters.
+
+```TypeScript
+copyFrom(sourceRange: Range | string, copyType?: "All" | "Formulas" | "Values" | "Formats", skipBlanks?: boolean, transpose?: boolean): void;
+```
+
+`copyType` specifies what data gets copied from the source to the destination.
+
+- `"Formulas"` transfers the formulas in the source cells and preserves the relative positioning of those formulas’ ranges. Any non-formula entries are copied as-is.
+- `"Values"` copies the data values and, in the case of formulas, the result of the formula.
+- `"Formats"` copies the formatting of the range, including font, color, and other format settings, but no values.
+- `"All"` (the default option) copies both data and formatting, preserving cells’ formulas if found.
+
+`skipBlanks` sets whether blank cells are copied into the destination. When true, `copyFrom` skips blank cells in the source range. 
+Skipped cells will not overwrite the existing data of their corresponding cells in the destination range. The default is false.
+
+The following code sample and images demonstrate this behavior in a simple scenario.
+
+```js
+Excel.run(function (context) {
+    var sheet = context.workbook.worksheets.getItem("Sample");
+    // copy a range, omitting the blank cells so existing data is not overwritten in those cells
+    sheet.getRange("D1").copyFrom("A1:C1",
+        Excel.RangeCopyType.all,
+        true, // skipBlanks
+        false); // transpose
+    // copy a range, including the blank cells which will overwrite existing data in the target cells
+    sheet.getRange("D2").copyFrom("A2:C2",
+        Excel.RangeCopyType.all,
+        false, // skipBlanks
+        false); // transpose
+    return context.sync();
+}).catch(errorHandlerFunction);
+```
+
+*Before the preceding function has been run.*
+
+![Data in Excel before range’s copy method has been run](../images/excel-range-copyfrom-skipblanks-before.png)
+
+*After the preceding function has been run.*
+
+![Data in Excel after range’s copy method has been run](../images/excel-range-copyfrom-skipblanks-after.png)
+
+`transpose` determines whether or not the data is transposed, meaning its rows and columns are switched, into the source location.
```

Comment:
Sorry I missed this the first time around -- suggest that you move these two sentences about the `transpose` parameter to immediately before the sentence that introduces the code sample "The following code sample..." (i.e., the same location where you describe the other two parameters).